schema-registry/client/src/main/java/com/hortonworks/registries/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -619,19 +702,25 @@ else if (cause instanceof IncompatibleSchemaException) {
     public void deleteSchemaVersion(SchemaVersionKey schemaVersionKey) throws SchemaNotFoundException, SchemaLifecycleException {
         schemaVersionInfoCache.invalidateSchema(new SchemaVersionInfoCache.Key(schemaVersionKey));
 
-        WebTarget target = currentSchemaRegistryTargets().schemasTarget.path(String.format("%s/versions/%s", schemaVersionKey
-                .getSchemaName(), schemaVersionKey.getVersion()));
-        Response response = null;
-        try {
-            response = login.doAction(new PrivilegedAction<Response>() {
-                @Override
-                public Response run() {
-                    return target.request(MediaType.APPLICATION_JSON_TYPE).delete(Response.class);
-                }
-            });
-        } catch (LoginException e) {
-            throw new RegistryRetryableException(e);
-        }
+        Response response = RETRY_MANAGER.execute(new RetryContext.Builder<Response>().policy(retryPolicy)
```

Comment:
Building the RetryContext with same policy and retryOnException for all the requests is redundant. Can you extract it in a method?

(eg)
```java
   @Override
    public Collection<SchemaVersionKey> findSchemasByFields(SchemaFieldQuery schemaFieldQuery) {
        return invoke((SchemaRegistryTargets targets) -> {
            WebTarget target = targets.searchFieldsTarget;
            for (Map.Entry<String, String> entry : schemaFieldQuery.toQueryMap().entrySet()) {
                target = target.queryParam(entry.getKey(), entry.getValue());
            }
            return getEntities(target, SchemaVersionKey.class);
        });
    }

    private <T> T invoke(RetryBody<T> retryBody) {
        return RETRY_MANAGER.execute(new RetryContext.Builder<T>()
                .policy(retryPolicy)
                .retryOnException(RegistryRetryableException.class)
                .build(() -> runWithFailoverContext(retryBody)));
    }
```

Reply:
I was thinking about removing Request class from RetryContext, then retryManager can be invoked as 
`retryManager.execute(retryContext, () -> {}); `
This translates to execute this block code against this retryContext, which makes sense.
That along with your invoke method would clean up the code nicely.